src/librustc_typeck/astconv.rs

```diff
@@ -565,6 +577,7 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
             tcx.sess.span_err(binding.span, &msg);
         }
         tcx.check_stability(assoc_ty.def_id, Some(ref_id), binding.span);
+        dup_bindings.entry(assoc_ty.def_id).or_insert(Vec::new()).push(binding.span);
```

Comment:
Could we model this error after duplicated fields:
```
error[E0062]: field `a` specified more than once
 --> src/main.rs:4:23
  |
4 |     let s = S { a: 0, a: 1 };
  |                 ----  ^ used more than once
  |                 |
  |                 first use of `a`
```

We keep `FxHashMap<DefId, Span>` instead of `FxHashMap<DefId, Vec<Span>>` and if the inserted `DefId` was already in the table, report an error pointing to the first use right here.
I'd even reuse all the wording for fields (except for replacing "field `a`" with "associated type binding `a`", of course).

Reply:
(You can attach labels and notes to the object returned by `struct_span_lint_node`, there are plenty of examples in the codebase.)

Reply:
Fixed.